unified-runtime/source/adapters/level_zero/device.cpp

```diff
@@ -167,7 +167,7 @@ ur_result_t urDeviceGet(
     std::copy_n(MatchedDevices.begin(), N, Devices);
 
   if (NumDevices) {
```

Comment:
Shouldn't this be just `if (NumDevices) *NumDevices = ZeDeviceCount;`?

Right now, we check the same condition twice (line 169 and 170). The code on line 173 will never be executed.